arangod/Aql/Functions.cpp

```diff
@@ -3512,6 +3512,129 @@ AqlValue Functions::Average(arangodb::aql::Query* query,
   return AqlValue(AqlValueHintNull());
 }
 
+
+#define VPA(a, b) VPackArrayBuilder guard##a(b.get())
+#define VPO(a, b) VPackObjectBuilder guard##a(b.get())
+AqlValue Functions::TestInternal(arangodb::aql::Query* query,
+                                 transaction::Methods* trx,
+                                 VPackFunctionParameters const& parameters) {
+  char const* AFN = "TEST_INTERNAL";
+  ValidateParameters(parameters, AFN, 2, 2);
+  
+  AqlValue atest = ExtractFunctionParameterValue(parameters, 0);
+  AqlValue awhat = ExtractFunctionParameterValue(parameters, 1);
+
+  if (!atest.isString()) {
+    RegisterInvalidArgumentWarning(query, AFN);
+    return AqlValue(AqlValueHintNull());
+  }
+  
+  std::string testName = atest.slice().copyString();
+  if (testName == "MODIFY_ARRAY") {
+    if (!awhat.isArray()) {
+      RegisterInvalidArgumentWarning(query, AFN);
+      return AqlValue(AqlValueHintNull());
+    }
+
+    AqlValueMaterializer materializer(trx);
+    VPackSlice what = materializer.slice(awhat, false);
+
+    transaction::BuilderLeaser smallArray(trx);  // just [1, 2]
+    { VPA(1, smallArray);
```

Comment:
I fail to see the advantage over:

    smallArray->openArray();
    smallArray->add(VPackValue(1));
    smallArray->add(VPackValue(2));
    smallArray->close();

same amount of lines - no macro